src/mdlComponentHandler.js

```diff
@@ -86,6 +86,15 @@ var componentHandler = (function() {
    * the element to.
    */
   function upgradeElementInternal(element, jsClass) {
+    if (jsClass === undefined) {
+      for (var i = 0; i < registeredComponents_.length; i++) {
```

Comment:
Maybe I am being oldschool and maybe new browsers dont have this issue, but I was always under the impression that foreach on arrays was like 95% slower than for. I wouldnt want IE9 to become slow or something even if it does support it. Eg see here: https://coderwall.com/p/kvzbpa/don-t-use-array-foreach-use-for-instead